### PR TITLE
SystemInfo: RawDeviceId creation: Ethernet card name configuration provision added

### PR DESCRIPTION
**Pull request by HaseenaSainul**
1. Moved RawDeviceId API from Core to WPEFramework, to limit its usage inside it. And try to ensure Network subsystem is up before trying to read MAC Ids
2. Added Ethernet Card name configuration

### CODE REVIEW

File: Source/WPEFramework/PluginHost.cpp [hunk 1]
```diff
@@ -545,7 +545,7 @@ namespace PluginHost {
             SYSLOG(Logging::Startup, (_T(EXPAND_AND_QUOTE(APPLICATION_NAME))));
             SYSLOG(Logging::Startup, (_T("Starting time: %s"), Core::Time::Now().ToRFC1123(false).c_str()));
             SYSLOG(Logging::Startup, (_T("Process Id:    %d"), Core::ProcessInfo().Id()));
-            SYSLOG(Logging::Startup, (_T("SystemId:      %s"), Core::SystemInfo::Instance().Id(Core::SystemInfo::Instance().RawDeviceId(), ~0).c_str()));
+            SYSLOG(Logging::Startup, (_T("SystemId:      %s"), Core::SystemInfo::Instance().Id(SystemInfo::RawDeviceId(_config->EthernetCard()), ~0).c_str()));
```
Comment (pwielders) [inline Source/WPEFramework/PluginHost.cpp]: We should now get this information only through the ISubSystem->IIdentifier interface. The SystemInfo::RawDeviceId, should dissapear as a visible interface.

File: Source/WPEFramework/SystemInfo.h [hunk 2]
```diff
@@ -373,8 +375,17 @@ namespace PluginHost {
                         _identifier->Release();
                     }
 
+                    uint32_t timeLeft = WaitTimeForInterfaceReady;
```
Comment (pwielders) [inline Source/WPEFramework/SystemInfo.h]: Lets not wait for an interface to become available. If we need to do so we should do it in the PluginHost.cpp (where it is already waiting for the localhost interface) But if we do it here we delay the full startup of all the other plugins.

File: Source/WPEFramework/SystemInfo.h [hunk 3]
```diff
@@ -27,11 +27,124 @@ namespace WPEFramework {
 namespace PluginHost {
 
     class SystemInfo : public PluginHost::ISubSystem {
-    private:
+    public:
         SystemInfo() = delete;
         SystemInfo(const SystemInfo&) = delete;
         SystemInfo& operator=(const SystemInfo&) = delete;
 
+    private:
```
Comment (pwielders) [inline Source/WPEFramework/SystemInfo.h:35]: Looks like this is a copy of the core, why not just use the core functionality, why copy it here ?

**Review by pwielders — CHANGES_REQUESTED** [review]

File: Source/plugins/IPlugin.h [hunk 4]
```diff
@@ -223,6 +223,14 @@ namespace PluginHost {
         virtual ISecurity* Officer(const string& token) = 0;
     };
 
+    struct EXTERNAL IConfigure : virtual public Core::IUnknown {
```
Comment (pwielders) [inline Source/plugins/IPlugin.h]: There is already an IConfigure you can use for the plugins, no need to define a new one that is part of the core. See interfaces/IConfigure..

File: Source/com/Ids.h [hunk 5]
```diff
@@ -54,7 +55,7 @@ namespace RPC {
         ID_SUBSYSTEM_LOCATION = 0x00000036,
         ID_SUBSYSTEM_IDENTIFIER = 0x00000037,
         ID_SUBSYSTEM_TIME = 0x00000038,
-		ID_SUBSYSTEM_SECURITY = 0x00000039,
+        ID_SUBSYSTEM_SECURITY = 0x00000039,
```
Comment (pwielders) [inline Source/com/Ids.h:57]: Already exists as an interface in the ThunderInterfaces, no need to make this part of the Thunder.

Comment (HaseenaSainul) [reply Source/com/Ids.h:57]: Sure, will use that from interfaces

File: Source/plugins/IPlugin.h (hunk 4 above)
Comment (HaseenaSainul) [reply Source/plugins/IPlugin.h]: Sure will use that from interfaces

File: Source/WPEFramework/PluginHost.cpp (hunk 1 above)
Comment (HaseenaSainul) [reply Source/WPEFramework/PluginHost.cpp]: I think at this point system Identifier is not configured, so do we need to query for ISubSystem->IIdentifier

File: Source/WPEFramework/SystemInfo.h (hunk 3 above)
Comment (HaseenaSainul) [reply Source/WPEFramework/SystemInfo.h:35]: Moved this to DeviceIdentification plugin

File: Source/WPEFramework/PluginHost.cpp (hunk 1 above)
Comment (HaseenaSainul) [reply Source/WPEFramework/PluginHost.cpp]: Moved this after _dispatcher->Open as per our discussion

File: Source/WPEFramework/SystemInfo.h (hunk 2 above)
Comment (HaseenaSainul) [reply Source/WPEFramework/SystemInfo.h]: removed wait, as per the discussion now try for whatever ready in the deviceidentificaiton plugin not configured case.

File: Source/com/Ids.h (hunk 5 above)
Comment (HaseenaSainul) [reply Source/com/Ids.h:57]: done

File: Source/plugins/IPlugin.h (hunk 4 above)
Comment (HaseenaSainul) [reply Source/plugins/IPlugin.h]: done

File: Source/WPEFramework/PluginHost.cpp [hunk 6]
```diff
@@ -573,6 +595,11 @@ namespace PluginHost {
             // If we have handlers open up the gates to analyze...
             _dispatcher->Open();
 
+            string id = GetDeviceId(_dispatcher->Services().SubSystemsInterface());
```
Comment (pwielders) [inline Source/WPEFramework/PluginHost.cpp]: We are now losing a reference. the returned SubSystemsInterface() is an Addref'ed interface, it is never Released, so it is currently leaking. Move this call (SubSystemsInterface()) into the GetDeviceId method so you can release the interface, if it is given..

**Review by pwielders — CHANGES_REQUESTED** [review]
Interface leak on ISubSystem

File: Source/WPEFramework/PluginHost.cpp (hunk 6 above)
Comment (HaseenaSainul) [reply Source/WPEFramework/PluginHost.cpp]: done

**Review by pwielders — APPROVED** [review]

**Pull request merged**